### ISSUE TITLE
Amazon Stock shadow seems to mix in windmill

### ISSUE THREAD

**Issue by HiranChaudhuri** [paulwedeck/settlers-remake#54]
The Amazon stock shadow is not related to the building elements.
It looks more like the windmill animations were reused as shadows.

Here is some data about my system:
```
OS: Linux 5.13.0-48-generic
JVM: OpenJDK 64-Bit Server VM 16.0.1+9-Ubuntu-120.04
Settlers Remake: Workflow Run 195

```

**Comment by HiranChaudhuri** [comment]
![Screenshot from 2022-06-15 16-47-47](https://user-images.githubusercontent.com/6866487/173857380-61396f94-9e9b-4b9a-89b4-af03b9d1394d.png)

**Comment by paulwedeck** [comment]
Yes, it has the wrong shadow. There is a mechanism to correct the shadows. Just look at the ShadowMapping interface

**Comment by HiranChaudhuri** [comment]
Apparently, the DAT File viewer is no help for finding shadows. So I had to have a look at the exported graphics.

It seems the file found in graphics/43/settlers/014/000_shadow.png is the best match for the Amazon stock.
Likewise, the shadows for the mill are also wrong. Currently they remind me of the tower?

![Screenshot from 2022-06-16 08-32-09](https://user-images.githubusercontent.com/6866487/174006487-1ed72d81-034d-4adf-876c-e610361402aa.png)

And the tower - umm, yes.

![Screenshot from 2022-06-16 08-33-39](https://user-images.githubusercontent.com/6866487/174006698-2fd3f616-3dc4-4c05-97c8-2e9a47aabb4b.png)

It seems the building editor is proving it's value.

**Comment by HiranChaudhuri** [comment]
With that patch it is better than before.
Now it looks strange that
- the mill's shadow does not follow the animation
- the miss seems to spin backwards

How can that be fixed?

**Comment by paulwedeck** [comment]
search for MILL in MapObjectDrawer

**Comment by HiranChaudhuri** [comment]
It seems like [this line](https://github.com/paulwedeck/settlers-remake/blob/master/jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java#L1343) limits to displaying a single image as shadow.

Now regardless of what I tried to render the sequence images' shadows I still do not get a good result.
Hence I believe I need to better control the sequence and shadow loading mechanism - can you provide some help there?

**Comment by paulwedeck** [comment]
Has every frame of the mill animation a proper shadow? If yes, then you should just draw the images with draw and remove the shadow part.

**Comment by HiranChaudhuri** [comment]
No, just a few frames seem to have shadows attached. The animation is too fast for me to really see them.

**Comment by HiranChaudhuri** [comment]
I have extended the BuildingCreator with functionality so it can (rudimentally) show an animation.
Now it becomes evident that the mill shadows in the game do not rotate at all but they could/should.

For the roman mill it looks ok (although I am puzzled watching the show).
The asian one seems to rotate in contradiction to it's shadow.
The egyption is ok.
The amazon one is lacking shadows.
